quickwit-metastore/src/metastore_resolver.rs

```diff
@@ -22,16 +22,19 @@
 
 use std::collections::HashMap;
 use std::sync::Arc;
+use async_trait::async_trait;
 
+use crate::metastore::single_file_metastore::SingleFileMetastoreFactory;
 use crate::{Metastore, MetastoreResolverError};
 
 /// A metastore factory builds a [`Metastore`] object from an URI.
 #[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
+#[async_trait]
 pub trait MetastoreFactory: Send + Sync + 'static {
     /// Returns the protocol this URI resolver is serving.
     fn protocol(&self) -> String;
     /// Given an URI, returns a [`Metastore`] object.
```

Comment:
I change &str to String just to avoid compilation error related to async / lifetime stuff